### DISCUSSION TITLE
Entity creation with same identifier should always fail, also when already loaded in entitymanager

### DISCUSSION THREAD

**Discussion by nvanlo** [mikro-orm/mikro-orm#6306] [General]
### Describe the bug

When an entity is loaded in the entity manager, a creation of an entity with the same identifier does **NOT** throw an exception.

This is unexpected behavior and leads to inconsistent behavior as creating that same entity when it is not yet loaded in the entity manager will lead to a sql constraint violation error.

For reference, Hibernate throws a specific exception when persisting an entity with the same identifier as an already loaded entity: EntityExistsException

Example repo: https://github.com/nvanlo/mikroorm-repro-no-changeset

### Reproduction

https://github.com/nvanlo/mikroorm-repro-no-changeset

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.4

### Node.js version

22

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
That's a feature, not a bug. Changing this would be breaking.

**Comment by nvanlo** [comment]
To keep it backwards compatible and implement this change in a non breaking way, we can introduce a config flag to indicate the expected behavior in this case.

**Reply by B4nan** [reply]
Yeah for sure, I am trying to understand why you think this should be a problem. Do you realize `em.create()` is also used for creating managed entity instances? And this is not just about `em.create`, it's how the EntityFactory works internally, so it's about all the entities the ORM creates too. We always ensure identity, so if you provide a PK in the payload, you always get the managed instances if available.

I am not sure what exactly we should validate here.

**Reply by nvanlo** [reply]
We now see different behavior when persisting an entity with `em.create()`  in following cases:

1. When the resultCache in the EntityManager is empty, we get a `UniqueConstraintViolation` error (from the underlying SQL driver)
2. When the resultCache contains an entity with the same id as we want to create, no error is thrown

For me this is an unexpected inconsistency in behavior, it also works different in for instance Hibernate, Hibernate will throw an `EntityExistsException` instead of a `UniqueConstraintViolation`.
I created a small test repo with hibernate to illustrate this: https://github.com/nvanlo/hibernate-test

I would expect we always get an error when explicitly creating / persisting an instance of an entity with an id that already exists.
